spec/requests/api/repositories_spec.rb

```diff
@@ -113,4 +113,38 @@
       end
     end
   end
+
+  describe "GET /api/github/:owner/:name/sync", type: :request do
+    context "without internal api key" do
+      it "forbids action" do
+        get "/api/github/#{repository.full_name}/sync", params: { api_key: create(:user).api_key }
```

Comment:
What happens with a nil or missing api_key?

Reply:
just added a spec above for that case